File: scripts/fix-routes.js
#!/usr/bin/env node

import { readFileSync, writeFileSync } from 'fs';
import { resolve } from 'path';

const servicesDir = resolve(process.cwd(), 'apps/web/src/services');

// Leer todos los archivos .ts en services
const { readdirSync } = await import('fs');
const files = readdirSync(servicesDir).filter(f => f.endsWith('.ts'));

files.forEach(file => {
  const filePath = resolve(servicesDir, file);
  let content = readFileSync(filePath, 'utf-8');
  
  // Reemplazar rutas con slash inicial - múltiples pasadas
  let prevContent;
  do {
    prevContent = content;
    // api.get('/admin → api.get('admin
    content = content.replace(/\.get\(['"]\//g, ".get('");
    content = content.replace(/\.post\(['"]\//g, ".post('");
    content = content.replace(/\.put\(['"]\//g, ".put('");
    content = content.replace(/\.delete\(['"]\//g, ".delete('");
    content = content.replace(/\.patch\(['"]\//g, ".patch('");
    // api.get(`/admin → api.get(`admin
    content = content.replace(/\.get\(`\//g, ".get(`");
    content = content.replace(/\.post\(`\//g, ".post(`");
    content = content.replace(/\.put\(`\//g, ".put(`");
    content = content.replace(/\.delete\(`\//g, ".delete(`");
    content = content.replace(/\.patch\(`\//g, ".patch(`");
  } while (content !== prevContent);
  
  writeFileSync(filePath, content, 'utf-8');
  console.log(`✅ ${file} actualizado`);
});

console.log('\n🎉 Todos los servicios actualizados');

File: scripts/fix-routes.py
#!/usr/bin/env python3
import os
import re

services_dir = 'apps/web/src/services'

for filename in os.listdir(services_dir):
    if not filename.endswith('.ts'):
        continue
    
    filepath = os.path.join(services_dir, filename)
    
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Reemplazar rutas con slash inicial - maneja tipos genéricos
    # api.get<{...}>('/admin → api.get<{...}>('admin
    content = re.sub(r"\.get<[^>]*>\('\/", lambda m: m.group(0).replace("('/", "('"), content)
    content = re.sub(r"\.post<[^>]*>\('\/", lambda m: m.group(0).replace("('/", "('"), content)
    content = re.sub(r"\.put<[^>]*>\('\/", lambda m: m.group(0).replace("('/", "('"), content)
    content = re.sub(r"\.delete<[^>]*>\('\/", lambda m: m.group(0).replace("('/", "('"), content)
    content = re.sub(r"\.patch<[^>]*>\('\/", lambda m: m.group(0).replace("('/", "('"), content)
    
    # También para template strings
    content = re.sub(r"\.get<[^>]*>\(`\/", lambda m: m.group(0).replace("(`/", "(`"), content)
    content = re.sub(r"\.post<[^>]*>\(`\/", lambda m: m.group(0).replace("(`/", "(`"), content)
    content = re.sub(r"\.put<[^>]*>\(`\/", lambda m: m.group(0).replace("(`/", "(`"), content)
    content = re.sub(r"\.delete<[^>]*>\(`\/", lambda m: m.group(0).replace("(`/", "(`"), content)
    content = re.sub(r"\.patch<[^>]*>\(`\/", lambda m: m.group(0).replace("(`/", "(`"), content)
    
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)
    
    print(f'✅ {filename} actualizado')

print('\n🎉 Todos los servicios actualizados')
